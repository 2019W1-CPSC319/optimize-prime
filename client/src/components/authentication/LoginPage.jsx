import React, { Component } from 'react';
import { withRouter } from 'react-router';
import { withStyles } from '@material-ui/core/styles';
import clsx from 'clsx';

import company from '../../images/galvanize_long.png';
import galvanize from '../../images/galvanize.png';
import microsoft from '../../images/microsoft.png';
import {
  Typography,
} from '@material-ui/core';

const styles = {
  button: {
    borderRadius: '5px',
    color: '#ffffff',
    display: 'flex',
    padding: '5px 10px',
    margin: '10px auto',
  },
  company: {
    display: 'block',
    margin: '200px auto 50px',
    width: '300px',
  },
  divider: {
    height: '35px',
    width: '1px',
    backgroundColor: '#ffffff',
    margin: 'auto 10px',
  },
  guest: {
    backgroundColor: '#000',
    padding: '15px 0',
    width: '320px',
  },
  logo: {
    width: '25px',
    margin: 'auto',
  },
  microsoft: {
    backgroundColor: '#3f51b5',
  },
  text: {
    margin: 'auto 50px',
  },
  noIconText: {
    margin: 'auto',
  },
  help: {
    textAlign: 'center',
    fontSize: '12px'
  },
  highlight: {
    color: '#483d8b',
    marginLeft: '10px',
  },
};

class LoginPage extends Component {
  constructor(props) {
    super(props);
    this.state = {

    };
  }

  render() {
    const { classes } = this.props;

    const signinLocation = window.location.host + "/auth/signin";

    return (
      <div>
        <div >
          <img className={classes.company} src={company} alt={'Galvanize Logo'} />
          <button className={`${classes.button} ${classes.microsoft}`} onClick={() => window.location.assign("/auth/signin")}>
            <img src={microsoft} className={classes.logo} alt={'Microsoft Logo'} />
            <div className={classes.divider}></div>
            <p className={classes.text}>Sign in with Microsoft Account</p>
          </button>
          <Typography align="center">or</Typography>
          <button className={clsx(classes.button, classes.guest)}>
            <p className={classes.noIconText}>Sign in with Guest Account</p>
          </button>
        </div>
      </div>
    );
  }
}

export default withStyles(styles)(withRouter(LoginPage));
