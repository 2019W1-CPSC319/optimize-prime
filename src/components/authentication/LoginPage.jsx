import React, { Component } from 'react';
import { withRouter } from 'react-router';
import { withStyles } from '@material-ui/core/styles';
import company from '../../images/galvanize_long.png';
import galvanize from '../../images/galvanize.png';
import microsoft from '../../images/microsoft.png';
import {
  Typography,
} from '@material-ui/core';

const styles = {
  button: {
    borderRadius: '5px',
    color: '#ffffff',
    display: 'flex',
    padding: '5px 10px',
    margin: '10px auto',
  },
  company: {
    display: 'block',
    margin: '200px auto 50px',
    width: '300px',
  },
  divider: {
    height: '35px',
    width: '1px',
    backgroundColor: '#ffffff',
    margin: 'auto 10px',
  },
  galvanize: {
    backgroundColor: '#280e3a',
  },
  logo: {
    width: '25px',
    margin: 'auto',
  },
  microsoft: {
    backgroundColor: '#3f51b5',
  },
  text: {
    margin: 'auto 50px',
  },
  help: {
    textAlign: 'center',
    fontSize: '12px'
  },
  highlight: {
    color: '#483d8b',
    marginLeft: '10px',
  },
  footer: {
    color: '#ffffff',
    backgroundColor: '#333',
    position: 'absolute',
    bottom: '0',
    width: '100%',
    padding: '10px 0',
  },
  license: {
    padding: '0 25px',
  }
};

class LoginPage extends Component {
  constructor(props) {
    super(props);
    this.state = {

    };
  }

  render() {
    const { classes } = this.props;
    return <div>
      <div>
        <img className={classes.company} src={company} alt={'Galvanize Logo'}></img>
        <button className={`${classes.button} ${classes.microsoft}`}>
          <img src={microsoft} className={classes.logo} alt={'Microsoft Logo'}></img>
          <div className={classes.divider}></div>
          <p className={classes.text}>Sign in with Microsoft Account</p>
        </button>
        <Typography align="center">or</Typography>
        <button className={`${classes.button} ${classes.galvanize}`}>
          <img src={galvanize} className={classes.logo} alt={'Galvanize Logo'}></img>
          <div className={classes.divider}></div>
          <p className={classes.text}>Sign in with Galvanize Account</p>
        </button>
      </div>
    </div>;
  }
}

export default withStyles(styles)(withRouter(LoginPage));
