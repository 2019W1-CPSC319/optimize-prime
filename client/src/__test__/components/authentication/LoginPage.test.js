import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LoginPage from '../../../components/authentication/LoginPage';

describe('Login Page - Render', () => {
  let wrapper;
  const classes = {};
  const actions = {};

  beforeEach(() => {
    wrapper = shallow(<LoginPage
      classes={classes}
      actions={actions}
    />);
  });

  it('render the component', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
