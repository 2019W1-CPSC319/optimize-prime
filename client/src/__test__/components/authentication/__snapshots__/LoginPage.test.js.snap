// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page - Render match snapshot 1`] = `
<withRouter(LoginPage)
  actions={Object {}}
  classes={
    Object {
      "button": "withRouter(LoginPage)-button-1",
      "center": "withRouter(LoginPage)-center-7",
      "company": "withRouter(LoginPage)-company-2",
      "divider": "withRouter(LoginPage)-divider-3",
      "logo": "withRouter(LoginPage)-logo-4",
      "microsoft": "withRouter(LoginPage)-microsoft-5",
      "text": "withRouter(LoginPage)-text-6",
    }
  }
/>
`;
